engine/cli/commands/engine_install_cmd.cc

```diff
@@ -1,16 +1,82 @@
 #include "engine_install_cmd.h"
+#include "server_start_cmd.h"
+#include "utils/download_progress.h"
+#include "utils/engine_constants.h"
 #include "utils/logging_utils.h"
+#include "utils/json_helper.h"
 
 namespace commands {
+namespace {
+std::string NormalizeEngine(const std::string& engine) {
+  if (engine == kLlamaEngine) {
+    return kLlamaRepo;
+  } else if (engine == kOnnxEngine) {
+    return kOnnxRepo;
+  } else if (engine == kTrtLlmEngine) {
+    return kTrtLlmRepo;
+  }
+  return engine;
+};
+}  // namespace
 
-void EngineInstallCmd::Exec(const std::string& engine,
+bool EngineInstallCmd::Exec(const std::string& engine,
                             const std::string& version,
                             const std::string& src) {
-  auto result = engine_service_.InstallEngine(engine, version, src);
-  if (result.has_error()) {
-    CLI_LOG(result.error());
-  } else if(result && result.value()){
-    CLI_LOG("Engine " << engine << " installed successfully!");
+  // Handle local install, if fails, fallback to remote install
+  if (!src.empty()) {    
+    auto res = engine_service_.UnzipEngine(engine, version, src);
+    if(res.has_error()) {
+      CLI_LOG(res.error());
+      return false;
+    }
+    if(res.value()) {
+      CLI_LOG("Engine " << engine << " installed successfully!");
+      return true;
+    }
+  }
+
+  // Start server if server is not started yet
+  if (!commands::IsServerAlive(host_, port_)) {
+    CLI_LOG("Starting server ...");
+    commands::ServerStartCmd ssc;
+    if (!ssc.Exec(host_, port_)) {
+      return false;
+    }
+  }
+
+  httplib::Client cli(host_ + ":" + std::to_string(port_));
+  Json::Value json_data;
+  auto data_str = json_data.toStyledString();
+  cli.set_read_timeout(std::chrono::seconds(60));
+  auto res = cli.Post("/v1/engines/install/" + engine, httplib::Headers(),
+                      data_str.data(), data_str.size(), "application/json");
+
+  if (res) {
+    if (res->status == httplib::StatusCode::OK_200) {
```

Comment:
if (res->status != httplib::StatusCode::OK_200) is better?

Reply:
Updated, thank you.